### ISSUE TITLE
Maybe a bug with a modal

### ISSUE THREAD

**Issue by frederikhors** [bestguy/sveltestrap#105]
I don't know if this problem is with `svelte-spa-router` or with `sveltestrap`.

### Reproduction here: https://codesandbox.io/s/modern-cache-znqt3

But codesandbox is not the best place for routing issues.
You can download project: "File > Export to Zip".

### The problem

Reproduction steps:

1. Click on "Hello" menu item
1. Click on button
1. Modal should appear
1. Click on the back button of the browser
1. The problem is "Hello Home" and button are still there (on codesandbox the modal too).

![image](https://user-images.githubusercontent.com/41120635/71774024-b4d1b280-2f67-11ea-9b68-7b354c744067.png)

**Comment by bestguy** [comment]
Thanks, I'm seeing an error when I try and close the modal too, maybe the root cause?:

```[Error] Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'outros.c')
	transition_out (index.js:700)
	outro (HelloHome.svelte:138)
	hs (Anonymous Script 1 (line 1:4515))
       ...
```

**Comment by bestguy** [comment]
Actually disregard on above, when you download the codesandbox zip and run locally closing seems fine.  I do see the duplication error you see though.  Not sure cause, looking at it locally.

**Comment by bestguy** [comment]
@frederikhors it looks like issue is you have two Routers, one in App.svelte and one in Home.svelte:

```
<!-- <Router {routes} {prefix} /> -->
<Router {routes} />
```

If I remove the router code from Home.svelte this works fine.  Can't seem to fork the codesandbox to demo? but let me know if you think this is still an issue.

**Comment by frederikhors** [comment]
Yes. Two router is the way to go according to: https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Advanced%20Usage.md#nested-routers.

One in App.svelte is commented.

The one in Home.svelte is necessary.

**Comment by bestguy** [comment]
Depending on what you are trying to do, a single router seems to work:

App.svelte:
```
<script>
  import Router from "svelte-spa-router";
  import Hello from "./Hello.svelte";
  import HelloHome from "./HelloHome.svelte";

  const routes = {
    "/": Hello,
    "/hello": HelloHome
  };
</script>

<a href="#/">Home</a> | <a href="#/hello">HelloHome</a>
<Router {routes} />
```

Hello.svelte:
```
<script>
  import { onMount, onDestroy } from "svelte";

  onMount(() => {
    console.log("onMount() Hello.svelte");
  });

  onDestroy(() => {
    console.log("onDestroy() Hello.svelte");
  });
</script>

<h1>Hello</h1>
```

Not familiar enough with this router to say if it's something we have an issue with 🤷‍♂

**Comment by frederikhors** [comment]
Yes. I know. But this is just a little example. In my real `Hello.svelte` I have many routes (also fetched from back-end).

So this is just an example and it doesn't work.

I opened this issue because if I remove the `Modal` changing it with let's say a text it works with no problems at all:

```svelte
<button on:click={handleClick}>Bill Gates</button>

<!-- <Modal isOpen={isModalOpen} toggle={toggleModal}>
  <ModalHeader toggle={toggleModal}>Sample</ModalHeader>
  <ModalBody>
    Hello from modal!
  </ModalBody>
</Modal> -->

{#if isModalOpen}Bye bye Modal{/if}
```

Really I don't know what to think.
